public/javascripts/case-progress.js

```diff
@@ -3,18 +3,16 @@
   // The timer before saving the draft note after the user paused typing
   const debounceTimer = 1000;
 
-  function getAutoSaveHandler(textarea) {
-    const hearingId = textarea.dataset.hearingid
+  function getAutoSaveHandler(textarea, url, inputDataFormatter) {
     let timeoutId;
-
     const noteEventListener = (event) => {
       // If a timer was already started, clear it.
       if (timeoutId) clearTimeout(timeoutId);
       // Set timer that will save comment when it fires.
       timeoutId = setTimeout(function () {
         // Make ajax call to save data.
         const xhr = new XMLHttpRequest()
```

Comment:
just a curious point, but we can't use a js library I presume on the frontend? Also, I take it we can't support the new ```fetch``` method?

Reply:
We discussed among team about that but we're not sure we have used 3rd party libraries in MoJ and we were supposed to find that out but that didn't happen and I went with what other projects were doing

Reply:
So I know in HMCTS (my previous team at MoJ) we used Angular (https://github.com/hmcts/rpx-xui-webapp)

Reply:
Ah, that's great.